helix/transforms/tigress/tigress.py

```diff
@@ -0,0 +1,166 @@
+import json
+import magic
+import os
+import shutil
+from urllib import parse, request as req
+
+from json.decoder import JSONDecodeError
+
+from ... import utils
+from ... import transform
+from ... import exceptions
+
+
+class TigressError(Exception):
+    """Raised when Tigress fails."""
+
+    pass
+
+
+class TigressDependency(utils.Dependency):
+    """Using the Tigress Transform"""
+
+    def __init__(self, name):
+        if os.name != "posix":
+            raise exceptions.ConfigurationError(
+                "unsupported platform for this dependency type: {}".format(os.name)
+            )
+
+            self.name = name
+
+    def install(self, verbose):
+        """Downloading Tigress binaries and adding execution permission."""
+        print("  By installing, you accept the Tigress End-User License Agreement.")
+        url = "http://tigress.cs.arizona.edu/cgi-bin/projects/tigress/download.cgi"
+        data = {
+            "accept": "Accept and Download",
+            "mode": "download",
+            "buffer": "address:----email:----file:tigress-3.1-bin.zip----name:----remote_addr=----timestamp:----",
+            "file": "tigress-3.1-bin.zip",
+            "destfile": "tigress-3.1-bin.zip",
+        }
+        data = parse.urlencode(data).encode()
+        request = req.Request(url=url, data=data)
+        response = req.urlopen(request)
+
+        destination = os.path.expanduser("~/bin")
+        if not os.path.exists(destination):
+            os.makedirs(destination)
+
+        temp = destination + "/tigress-3.1-bin.zip"
+
+        open(temp, "wb").write(response.read())
+        shutil.unpack_archive(temp, destination)
+        os.remove(temp)
+
+        exec_permission = "chmod -R u+x " + destination + "/tigress"
+        utils.run(exec_permission, propagate=True)
+
+    def installed(self):
+        """Checks if Tigress is installed by guessing the path to the binary."""
+        binary = utils.find(
+            "tigress", guess=[os.path.expanduser("~/bin/tigress/3.1/tigress")]
+        )
+
+        return binary is not None
+
+
+class TigressTransform(transform.Transform):
+    """Transform for the Tigress C Diversifier/Obfuscator."""
+
+    name = "tigress"
+    verbose_name = "Tigress"
+    description = "The Tigress Diversifier/Obfuscator (v3.1)."
+    version = "1.0.0"
+    type = transform.Transform.TYPE_SOURCE
+
+    dependencies = [TigressDependency("tigress")]
+
+    options = {
+        "Recipe": {"default": "simple-recipe.json"},
+    }
+
+    def supported(self, source):
+        """Checks if Tigress supports the given file.
+
+        Verifies that the target file has a .c file extension.
```

Comment:
Is this docstring from an earlier iteration? Seems like you're using filemagic now, not checking the file extension

Reply:
Yeah - nice catch.